### ISSUE TITLE
Initial seleceted image after 'update' is second in order.

### ISSUE THREAD

**Issue by songoo** [schalkt/angular-super-gallery#12]
Hello.
Really great gallery! Super B :)

When i reload images with :

```
                $rootScope.$broadcast('ASG-gallery-edit', {
                    id: 'productImages',
                    update: thisInCustomVariable.productImages
                });
```

Gallery load new images all right, but for some reason, the second image gets selected, not the first one.

Screen just after images get updated/loaded - https://imgur.com/a/NjJ6Gdv

Before update, images array is empty array.

Is there maybe possibility to set which image should be selected/default/ticked by js ?

I also tried to call, 'refresh' but didnt help, still second image is as default.

Here is my current config 

```
        $scope.productImagesOptions = {
            "baseUrl": "/admin",
            "fields": {
                "source": {
                    "modal": "link",
                    "image": "medium",
                    "panel": "thumbnail"
                },
                title: 'title'
            },
            "loadingImage": "preload.svg",
            "preloadNext": true,
            "preloadDelay": 420,
            "theme": "darkblue",
            "thumbnail": {
                "height": 70,
                "index": true
            },
            "modal": {
                "caption": {
                    "visible": true,
                    "position": "bottom"
                },
                "header": {
                    "enabled": true,
                    "dynamic": false
                },
                "transition": "rotateLR",
                "title": "",
                "subtitle": "",
                "thumbnail": {
                    "height": 77,
                    "index": true
                },
                "size": 'contain'
            },
            "panel": {
                "click": {
                    "select": false,
                    "modal": true
                },
                "hover": {
                    "select": true
                },
                "item": {
                    "class": "custom",
                    "title": false
                },
                "arrows": {
                    "enabled": true, // show/hide arrows
                    "preload": true, // preload image on mouseover
                },
            },
            "image": {
                "height": 320,
                "click": {
                    "modal": true
                },
                "transition": "zlideLR",
                "placeholder": "panel",
                "arrows": {
                    "enabled": true,  // show/hide arrows
                    "preload": true, // preload image on mouseover
                },
            },
            duplicates: true,
        };
```

**Comment by songoo** [comment]
If I enclose '<asg-image' into 'data-ng-if', to wait till images data are loaded it works ok, and the first image is selected (with exact same code just added one ng-if )

Issue happens only when I initialize gallery in template first straight away with empty results (or any default) and then update is called with data. Or also when i initialize it upon refresh of page (its ok) and then by angular driven refresh to another page it again jumps on second image again.

**Comment by schalkt** [comment]
@songoo Thanks for your feedback and the great words :)

I fixed some bugs in the new version (v1.5.9) and added "selected" option to the event data.
So you can change the selected image directly when edit.

```javascript
	$rootScope.$broadcast('ASG-gallery-edit', {
		id: 'nature',
		selected: 0,
		update: this.files1,
		options: this.options1
	});
```

But you can easily manipulating the selected image from your code by the data-selected="..." attribute.
See in the code of demo page.

```html
<asg-image data-id="nature" data-selected="nature1Selected" data-options="ctrl.options1" data-items="ctrl.files1">
```
I hope this helps

**Comment by songoo** [comment]
Tnx much ! That was some fast response :P 

options: 0 , did definitely the trick :P , again cool gallery :P 

(Had some 2x accumulating images trouble with updating in same way, but 'duplicates:false' solved that matter :P )
